api/views.py

```diff
@@ -268,16 +269,17 @@ def _only_latest_date(self, iterable):
         if next_ is not None:
             yield next_
 
-    def listify(self, iterable, request, format=None):
+    def listify(self, iterable, request=None, format=None):
         """Some renderers require lists instead of any iterables for rendering.
         This function conditionally turns iterables into lists based on the
         format requested by browser."""
         # choose either '?format=...' or '/url.format' or None
-        format_ = (request.query_params.get('format') or format or '').lower()
+        if request is not None:
```

Comment:
Sorry, I probably sounded rude, but I actually meant to ask what was this code doing and why, because you hadn't included any explanation.